src/test/java/org/minions/devfund/katerinaanzoleaga/battleship/ShipTest.java

```diff
@@ -0,0 +1,280 @@
+        ship.shootAt(row, columnSubmarineOne);
+        ship.shootAt(row, columnSubmarineTwo);
+        ship.shootAt(row, columnSubmarineZero);
+        assertTrue(ship.isSunk());
+    }
+    /**
+     * Test several shots until sunk a vertical destroyer.
+     */
+    @Test
+    public void testSunkDestroyerH() {
+        final Ship destroyer = new Destroyer();
+        destroyer.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < destroyer.getLength(); ++i) {
+            destroyer.shootAt(0, i);
+            if (i != destroyer.getLength() - 1) {
+                assertFalse(destroyer.isSunk());
+            }
+        }
+        assertTrue(destroyer.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal destroyer.
+     */
+    @Test
+    public void testSunkDestroyerV() {
+        final Ship destroyer = new Destroyer();
+        destroyer.placeShipAt(0, 0, false, theOcean);
+        for (int i = 0; i < destroyer.getLength(); ++i) {
+            destroyer.shootAt(i, 0);
+            if (i != destroyer.getLength() - 1) {
+                assertFalse(destroyer.isSunk());
+            }
+        }
+        assertTrue(destroyer.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal lightCruiser.
+     */
+    @Test
+    public void testSunkLightCruiserH() {
+        final Ship lightCruiser = new Destroyer();
+        lightCruiser.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < lightCruiser.getLength(); ++i) {
+            lightCruiser.shootAt(0, i);
+            if (i != lightCruiser.getLength() - 1) {
+                assertFalse(lightCruiser.isSunk());
+            }
+        }
+        assertTrue(lightCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical lightcruiser.
+     */
+    @Test
+    public void testSunkLightCruiserV() {
+        final Ship lightCruiser = new LightCruiser();
+        assertFalse(lightCruiser.isSunk());
+        lightCruiser.placeShipAt(0, 0, false, theOcean);
+        for (int i = 0; i < lightCruiser.getLength(); ++i) {
+            lightCruiser.shootAt(i, 0);
+            if (i != lightCruiser.getLength() - 1) {
+                assertFalse(lightCruiser.isSunk());
+            }
+        }
+        assertTrue(lightCruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk an horizontal cruiser.
+     */
+    @Test
+    public void testSunkCruiserH() {
+        final Ship cruiser = new Cruiser();
+        cruiser.placeShipAt(0, 0, true, theOcean);
+        for (int i = 0; i < cruiser.getLength(); ++i) {
+            cruiser.shootAt(0, i);
+            if (i != cruiser.getLength() - 1) {
+                assertFalse(cruiser.isSunk());
+            }
+        }
+        assertTrue(cruiser.isSunk());
+    }
+
+    /**
+     * Test several shots until sunk a vertical cruiser.
+     */
+    @Test
+    public void testSunkCruiserV() {
+        final Ship cruiser = new Cruiser();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "cruiser" which hides the field declared at line 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)